File: 507. Perfect Number/solution.cpp
class Solution {
public:
  bool checkPerfectNumber(int num) {
    if (num == 1){
      return false;
    }

    int total = 1;
    for (int index = 2; index < (int) sqrt(num) + 1; index++){
      if (num % index == 0){
        total += index + num / index;
      }
    }
    
    return total == num;
  }
};

File: 507. Perfect Number/solution.py
class Solution(object):
  def checkPerfectNumber(self, num):
    """
    :type num: int
    :rtype: bool
    """
    if num == 1:
      return False

    total = 1
    for index in range(2, int(sqrt(num)) + 1):
      if num % index == 0:
        total += index + num / index
    
    return total == num